### PR TITLE
Fixes: #7 - don't escape username and password

### PR DESCRIPTION
**Pull request by davidjwilkins**

### CODE REVIEW

**Comment by aeneasr** [comment]
Thanks! Could you also please add a test?

File: connector.go [hunk 1]
```diff
@@ -81,6 +81,12 @@ func (c *SQLConnection) GetDatabase() *sqlx.DB {
 	}
 
 	var err error
+
+	userinfo := c.URL.User.Username()
+	if password, hasPassword := c.URL.User.Password(); hasPassword {
+		userinfo = userinfo + ":" + password
```
Comment (aeneasr) [inline connector.go]: Does this work when userinfo or password have a `:`?

Comment (davidjwilkins) [reply connector.go]: This works when password has a `:`, but fails when username does.  It looks like this is related to https://github.com/go-sql-driver/mysql/issues/591 - their DSN parser doesn't support it.

Comment (davidjwilkins) [reply connector.go]: to be clear, the existing implementation also fails when the username has a :, and this does not fix that.  this does fix when the username or password has certain other characters

Comment (aeneasr) [reply connector.go]: Ah, didn't know that, thanks!

**Pull request merged**